tfsdk/config.go

```diff
@@ -60,6 +60,10 @@ func (c Config) GetAttribute(ctx context.Context, path path.Path, target interfa
 }
 
 // PathMatches returns all matching path.Paths from the given path.Expression.
+//
+// If a parent path is null or unknown, which would prevent a full expression
```

Comment:
I think we need to clarify what `parent path` is. 

I'm reading this  method doc and pretending I don't know anything about the path libraries yet: the fact that I don't know what a `parent path` is, makes it difficult to understand the follow up `, which would prevent a full expression from matching, ...`

Reply:
I think understanding that, would also explain why returning the parent path is better than returning empty.

Reply:
From out of band discussion, the `Path` and `Expression` Go documentation will be expanded in a followup pull request to start really talking about how they relate to schemas and the underlying data.